sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/eventhubs/kafka/AzureEventHubsKafkaAutoConfigurationTests.java

```diff
@@ -182,7 +182,7 @@ void shouldConfigureWhenAnotherKafkaPropertiesProvided() {
     }
 
     @Test
-    void shouldNotOverrideKafkaPropertiesWithConnectionStringAutoConfiguration() {
+    void shouldNotOverrideConnectionStringPropertiesWithOAuth2AutoConfiguration() {
```

Comment:
I think what the case tries to cover is when `SASL_MECHANISM` set to `PLAIN`, the OAuth BPP won't process, so this specific case does two things with given `connection string`:
- Make sure the `KafkaPropertiesBeanPostProcessor` will work, so the sasl properties are set correctly
- Make sure the customizers from `AzureEventHubsKafkaOAuth2AutoConfiguration` won't work

so the method name could be something like shouldConfigureSaslPlainWhenGivenConnectionString

Reply:
Or maybe we can have a test class, called `AzureEventHubsKafkaConfigurationTests`, which is for cases with both AutoConfiguration classes enabled